src/zenml/integrations/mlflow/steps/mlflow_registry.py

```diff
@@ -146,6 +146,11 @@ def mlflow_register_model_step(
         metadata.zenml_pipeline_run_uuid = pipeline_run_uuid
     if metadata.zenml_workspace is None:
         metadata.zenml_workspace = zenml_workspace
+    if metadata.model_extra and "mlflow_run_id" in metadata.model_extra:
```

Comment:
I'm a little confused by this change, can you explain the intended behaviour of this code?

To me it seems like:
- We first check whether the `mlflow_run_id` exists in the `model_extra` dict
- If it doesn't exist, we do nothing. Why don't we add the key in this case?
- If it does exist, we have another `if` condition, but both the `if` and `else` branch do exactly the same thing: They overwrite the key with the mlflow run id.

It seems like both the `if` and `else` condition do the same thing: Overwrite `mlflow_run_id` in the `model_extra` dict, no matter if it's `None` or not.

And why do we need the actual check ahead if the key even exists and not just always set it?

Reply:
Yep looks good now!